pkg/storage/backend/s3/config.go

```diff
@@ -2,17 +2,38 @@ package s3
 
 import (
 	"flag"
+	"net/http"
+	"time"
 
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 )
 
+// HTTPConfig stores the http.Transport configuration for the s3 minio client.
+type HTTPConfig struct {
+	IdleConnTimeout       time.Duration `yaml:"idle_conn_timeout"`
+	ResponseHeaderTimeout time.Duration `yaml:"response_header_timeout"`
+	InsecureSkipVerify    bool          `yaml:"insecure_skip_verify"`
+
+	// Allow upstream callers to inject a round tripper
+	Transport http.RoundTripper `yaml:"-"`
+}
+
+// RegisterFlagsWithPrefix registers the flags for TSDB s3 storage with the provided prefix
+func (cfg *HTTPConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.DurationVar(&cfg.IdleConnTimeout, prefix+"s3.http.idle-conn-timeout", 90*time.Second, "The time an idle connection will remain idle before closing.")
+	f.DurationVar(&cfg.ResponseHeaderTimeout, prefix+"s3.http.response-header-timeout", 2*time.Minute, "The amount of time the client will wait for a servers response headers.")
+	f.BoolVar(&cfg.InsecureSkipVerify, prefix+"s3.http.insecure-skip-verify", false, "If enabled, the client will accept any certificate and hostname.")
+}
+
 // Config holds the config options for an S3 backend
 type Config struct {
 	Endpoint        string         `yaml:"endpoint"`
 	BucketName      string         `yaml:"bucket_name"`
 	SecretAccessKey flagext.Secret `yaml:"secret_access_key"`
 	AccessKeyID     string         `yaml:"access_key_id"`
 	Insecure        bool           `yaml:"insecure"`
+
+	HTTP HTTPConfig `yaml:"http_config"`
```

Comment:
I would remove the `_config` suffix. This also fixes the drift between CLI flags and YAML (because in the CLI flags it's `.http.`)

```suggestion
	HTTP HTTPConfig `yaml:"http"`
```

Reply:
👍